tests/unit/test_log_parser.py

```diff
@@ -141,3 +142,64 @@ def test_new_intercept_build(self):
 
         self.assertEqual(list(build_action.sources)[0], r'/tmp/a b.cpp')
         self.assertEqual(build_action.lang, 'c++')
+
+    def test_skip_preproc(self):
+        """
+        Compiler preprocessor actions should be marked as skipped.
+        """
+        preprocessor_actions = StringIO('''[
+            {"directory": "/tmp",
+            "command": "g++ /tmp/a.cpp -E /tmp/a.cpp",
+            "file": "/tmp/a.cpp" },
+            {"directory": "/tmp",
+            "command": "g++ /tmp/a.cpp -MT /tmp/a.cpp",
+            "file": "/tmp/a.cpp" },
+            {"directory": "/tmp",
+            "command": "g++ /tmp/a.cpp -MM /tmp/a.cpp",
+            "file": "/tmp/a.cpp" },
+            {"directory": "/tmp",
+            "command": "g++ /tmp/a.cpp -MF /tmp/a.cpp",
+            "file": "/tmp/a.cpp" },
+            {"directory": "/tmp",
+            "command": "g++ /tmp/a.cpp -M /tmp/a.cpp",
+            "file": "/tmp/a.cpp" }]
+        ''')
+        build_actions = \
+            log_parser.parse_compile_commands_json(preprocessor_actions,
+                                                   ParseLogOptions())
+        for build_action in build_actions:
+            self.assertTrue(build_action.skip)
+
+    def test_keep_compile_and_dep(self):
+        """ Keep the compile command if -MD is set.
```

Comment:
Why do we keep `-MD` ? If `-MD` is present there must be a a same compile command without the `-MD`. Typically the build systems first execute the dep generation and then they execute the real command which generates the .o.

Reply:
The `-MD` command does compilation (generates the .o file) together with the dependency generation if `-E` is not set, in that case you do not need a separate command to compile the source file.

Reply:
@martong do you have any other questions? I would keep the `-MD` because it does compilation if `-E` is not set.

Reply:
Okay.

Reply:
LGTM!